aws/resource_aws_emr_instance_fleet_test.go

```diff
@@ -0,0 +1,551 @@
+			}
+		}
+
+		providerErr, ok := err.(awserr.Error)
+		if !ok {
+			return err
+		}
+
+		log.Printf("[ERROR] %v", providerErr)
+	}
+
+	return nil
+}
+
+func testAccCheckAWSEmrInstanceFleetExists(n string, v *emr.InstanceFleet) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No task fleet id set")
+		}
+		meta := testAccProvider.Meta()
+		conn := meta.(*AWSClient).emrconn
+		instanceFleets, err := fetchAllEMRInstanceFleets(conn, rs.Primary.Attributes["cluster_id"])
+		if err != nil {
+			return fmt.Errorf("EMR error: %v", err)
+		}
+
+		fleet := findInstanceFleetById(instanceFleets, rs.Primary.ID)
+		if fleet == nil {
+			return fmt.Errorf("No match found for (%s)", n)
+		}
+		v = fleet
+		return nil
+	}
+}
+
+const testAccAWSEmrInstanceFleetBase = `
+data "aws_availability_zones" "available" {
+  # Many instance types are not available in this availability zone
+  exclude_zone_ids = ["usw2-az4"]
+  state            = "available"
+  filter {
+    name   = "opt-in-status"
+    values = ["opt-in-not-required"]
+  }
+}
+resource "aws_vpc" "test" {
+  cidr_block           = "10.0.0.0/16"
+  enable_dns_hostnames = true
+  tags = {
+    Name = "tf-acc-test-emr-cluster"
+  }
+}
+resource "aws_internet_gateway" "test" {
+  vpc_id = aws_vpc.test.id
+  tags = {
+    Name = "tf-acc-test-emr-cluster"
+  }
+}
+resource "aws_security_group" "test" {
+  vpc_id = aws_vpc.test.id
+  ingress {
+    from_port = 0
+    protocol  = "-1"
+    self      = true
+    to_port   = 0
+  }
+  egress {
+    cidr_blocks = ["0.0.0.0/0"]
+    from_port   = 0
+    protocol    = "-1"
+    to_port     = 0
+  }
+  tags = {
+    Name = "tf-acc-test-emr-cluster"
+  }
+  # EMR will modify ingress rules
+  lifecycle {
+    ignore_changes = [ingress]
+  }
+}
+resource "aws_subnet" "test" {
+  availability_zone       = data.aws_availability_zones.available.names[0]
+  cidr_block              = "10.0.0.0/24"
+  map_public_ip_on_launch = false
+  vpc_id                  = aws_vpc.test.id
+  tags = {
+    Name = "tf-acc-test-emr-cluster"
+  }
+}
+resource "aws_route_table" "test" {
+  vpc_id = aws_vpc.test.id
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = aws_internet_gateway.test.id
+  }
+}
+resource "aws_route_table_association" "test" {
+  route_table_id = aws_route_table.test.id
+  subnet_id      = aws_subnet.test.id
+}
+resource "aws_iam_role" "emr_service" {
+  name = "%[1]s_default_role"
+  assume_role_policy = <<EOT
+{
+  "Version": "2008-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "elasticmapreduce.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOT
+}
+resource "aws_iam_role_policy_attachment" "emr_service" {
+  role       = aws_iam_role.emr_service.id
+  policy_arn = aws_iam_policy.emr_service.arn
```

Comment:
Similar to the change we made in the `aws_emr_cluster` testing, let's switch this to the AWS managed policy. 👍